Libs/vtkSegmentationCore/vtkSegmentation.cxx

```diff
@@ -1658,3 +2169,250 @@ void vtkSegmentation::DeserializeConversionParameters(std::string conversionPara
 {
   this->Converter->DeserializeConversionParameters(conversionParametersString);
 }
+
+//----------------------------------------------------------------------------
+int vtkSegmentation::GetNumberOfLayers(std::string representationName/*=""*/)
+{
+  if (representationName.empty())
+    {
+    representationName = this->MasterRepresentationName;
+    }
+
+  vtkNew<vtkCollection> layerObjects;
+  this->GetLayerObjects(layerObjects, representationName);
+  return layerObjects->GetNumberOfItems();
+}
+
+//----------------------------------------------------------------------------
+void vtkSegmentation::GetLayerObjects(vtkCollection* layerObjects, std::string representationName/*= ""*/)
+{
+  if (!layerObjects)
+    {
+    // TODO
+    return;
+    }
+  if (representationName.empty())
+    {
+    representationName = this->MasterRepresentationName;
+    }
+  layerObjects->RemoveAllItems();
+
+  int layerCount = 0;
+  std::set<vtkDataObject*> objects;
+  for (std::string segmentId : this->SegmentIds)
+    {
+    vtkSegment* segment = this->GetSegment(segmentId);
+    vtkDataObject* dataObject = segment->GetRepresentation(representationName);
+    if (dataObject && objects.find(dataObject) == objects.end())
+      {
+      objects.insert(dataObject);
+      layerObjects->AddItem(dataObject);
+      ++layerCount;
+      }
+    }
+}
+
+//----------------------------------------------------------------------------
+int vtkSegmentation::GetLayerIndex(std::string segmentId, std::string representationName/*=""*/)
+{
+  if (representationName.empty())
+    {
+    representationName = this->MasterRepresentationName;
+    }
+
+  vtkNew<vtkCollection> layerObjects;
+  this->GetLayerObjects(layerObjects, representationName);
+
+  vtkSegment* segment = this->GetSegment(segmentId);
+  if (!segment)
+    {
+    // TODO
```

Comment:
Please don't leave TODOs in the code without at least some explanation. This will convert into noise even for you after a few weeks, and is not useful for anybody else.

Reply:
Sorry, not intentional, I'll go through and make sure I haven't missed any TODO.

Reply:
No problem, this is what code reviews are for :)